scripts/e2e-run.sh

```diff
@@ -130,17 +201,35 @@ for TEST_FILE in "${TEST_FILES[@]}"; do
       SUCCESS=true
       echo "✅ Passed: $TEST_NAME (${DURATION}s, $ATTEMPT attempt(s))"
       echo
+      
+      # Stop recording (if recording was active)
+      if [ "$SHOULD_RECORD" = "true" ]; then
+        stop_recording "$DEBUG_OUTPUT"
+      fi
+      
       mv "$DEBUG_OUTPUT" "$ARTIFACTS_FOLDER/maestro/✅-$TEST_NAME-$ATTEMPT"
       break
     else
       END_TIME=$(date +%s)
       DURATION=$((END_TIME - START_TIME))
       echo "⚠️ Attempt $ATTEMPT failed for $TEST_NAME (${DURATION}s)"
       echo
+      
+      # Stop recording (if recording was active)
+      if [ "$SHOULD_RECORD" = "true" ]; then
+        stop_recording "$DEBUG_OUTPUT"
+      fi
+      
       mv "$DEBUG_OUTPUT" "$ARTIFACTS_FOLDER/maestro/❌-$TEST_NAME-$ATTEMPT"
+
+      # Enable recording for subsequent attempts after first failure
+      if [ "$ATTEMPT" -eq 1 ]; then
```

Comment:
Do we need this `if` at all? Like if we ended up in this path then probably we just want to start recording right?